api/src/main/java/org/openmrs/scheduler/tasks/AutoRetireUsersTask.java

```diff
@@ -0,0 +1,100 @@
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.scheduler.tasks;
+
+import org.apache.commons.lang.StringUtils;
+import org.openmrs.User;
+import org.openmrs.api.UserService;
+import org.openmrs.api.context.Context;
+import org.openmrs.util.DateUtil;
+import org.openmrs.util.OpenmrsConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A scheduled task that automatically retires users after the set number of days of inactivity. 
+ * The inactivity duration is set as a global property. 
+ * <a href="https://openmrs.atlassian.net/wiki/spaces/docs/pages/101318663/Creating+Auto-Deactivating+User+Task">Documentation</a>
+ * {@link OpenmrsConstants#GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS}
+ * 
+ * @since 2.7.0
+ */
+public class AutoRetireUsersTask extends AbstractTask {
+	
+	private static final Logger log = LoggerFactory.getLogger(AutoRetireUsersTask.class);
+	private static final String AUTO_RETIRE_REASON = "User retired due to inactivity";
+
+	/**
+	 * @see org.openmrs.scheduler.tasks.AbstractTask#execute()
+	 */
+	@Override
+	public void execute() {
+		if (!isExecuting) {
+			log.debug("Starting auto-retiring users");
+			
+			startExecuting();
+			
+			try {
+				UserService userService = Context.getUserService();
+				Set<User> usersToRetire = getUsersToRetire(userService);
+				
+				usersToRetire.forEach(user -> userService.retireUser(user, AUTO_RETIRE_REASON));
+			} catch (Exception e) {
+				log.error("Error occurred while auto-retiring users: ", e);
+			} finally {
+				log.debug("Auto-retiring users task ended");
+				stopExecuting();
+			}
+		}
+	}
+
+	Set<User> getUsersToRetire(UserService userService) {
+		final List<User> allUsers = userService.getAllUsers();
+		String numberOfDaysToRetire = Context.getAdministrationService().getGlobalProperty(OpenmrsConstants.GP_NUMBER_OF_DAYS_TO_AUTO_RETIRE_USERS);
+		
+		if (numberOfDaysToRetire == null || numberOfDaysToRetire.isEmpty()) {
+			return Collections.emptySet();
+		}
+		
+		long numberOfMillisecondsToRetire = DateUtil.daysToMilliseconds(Double.parseDouble(numberOfDaysToRetire));
+
+		return allUsers.stream()
+			.filter(user -> !user.isSuperUser() 
+				&& !user.isRetired() 
+				&& userInactivityExceedsDaysToRetire(user, numberOfMillisecondsToRetire)
+			)
+			.collect(Collectors.toSet());
+	}
+
+	boolean userInactivityExceedsDaysToRetire(User user, long numberOfMillisecondsToRetire) {
+		String lastLoginTimeString = Context.getUserService().getLastLoginTime(user);
+
+		if (StringUtils.isNotBlank(lastLoginTimeString)) {
+			long lastLoginTime = Long.parseLong(lastLoginTimeString);
+
+			boolean b = System.currentTimeMillis() - lastLoginTime >= numberOfMillisecondsToRetire;
```

Comment:
Since b does not look like a good variable name, why not just directly `return System.currentTimeMillis() - lastLoginTime >= numberOfMillisecondsToRetire;`?